packages/amplify-category-api/src/provider-utils/awscloudformation/pipeline-with-awaiter.ts

```diff
@@ -192,6 +192,43 @@ export class PipelineWithAwaiter extends Construct {
       } as Record<string, codebuild.BuildEnvironmentVariable>,
     );
 
+    const ecsDeployActionRole = new iam.Role(scope, 'EcsDeployActionRole', {
+      assumedBy: new iam.AccountRootPrincipal(),
+    });
+    ecsDeployActionRole.addToPolicy(
```

Comment:
How does this policy pick up the other required permissions (`ecs:DescribeServices`, `ecs:DescribeTaskDefinition`, etc) when you're only explicitly declaring `ecs:TagResource`?

Reply:
When we pass this custom role into the CDK construct's `EcsDeployAction`, it **adds** these additional policy statements that it requires to the role's policy document.

Reply:
Thanks, saw that in the CDK source that it's adding the policy statements to the incoming role